### PR TITLE
Registry replacer: Update dockerfiles from ocp-build-data

### PR DESCRIPTION
**Pull request by alvaroaleman**
The diff this PR produces is here: https://github.com/openshift/release/pull/10799

First commit is pure moving code around, second commit is the actual change.

### CODE REVIEW

File: cmd/registry-replacer/main.go [hunk 1]
```diff
@@ -509,3 +552,65 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
...
+}
+
+func updateDockerfilesToMatchOCPBuildData(
+	config *api.ReleaseBuildConfiguration,
+	promotionTargetToDockerfileMapping map[string]string,
+	majorMinorVersion string,
+	ignoredRepos sets.String,
+) {
+
+	// The tool only works for the current release
```
Comment (stevekuznetsov) [inline cmd/registry-replacer/main.go:575]: Technically we just need it to not be a branch that's being fast forwarded _to_ right? So like `release-4.2` would be fine

File: cmd/registry-replacer/main.go [hunk 2]
```diff
@@ -509,3 +552,65 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
...
+
+	for idx, image := range config.Images {
+		promotionTarget, ok := promotedTags[string(image.To)]
+		if !ok {
+			continue
+		}
+		stringifiedPromotionTarget := fmt.Sprintf("registry.svc.ci.openshift.org/%s/%s:%s", promotionTarget.Namespace, promotionTarget.Name, image.To)
+		dockerfilePath, ok := promotionTargetToDockerfileMapping[stringifiedPromotionTarget]
+		if !ok {
+			// TODO: Is "ocp build data doesn't know this" something we should handle?
```
Comment (stevekuznetsov) [inline cmd/registry-replacer/main.go:603]: If you promote e.g. a test image that is not productized that's fine I think

File: cmd/registry-replacer/main.go [hunk 3]
```diff
@@ -509,3 +552,65 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
...
+			continue
+		}
+		actualDockerFilePath := "Dockerfile"
+		if image.DockerfilePath != "" {
+			actualDockerFilePath = image.DockerfilePath
+		}
+		actualDockerFilePath = filepath.Join(image.ContextDir, actualDockerFilePath)
+		if dockerfilePath != actualDockerFilePath {
+			config.Images[idx].ContextDir = ""
```
Comment (stevekuznetsov) [inline cmd/registry-replacer/main.go:612]: This sets the working directory for the build, so if their Dockerfile uses paths relative to this, and we change it, that could be breaking

File: cmd/registry-replacer/main.go (hunk 1 above)
Comment (alvaroaleman) [reply cmd/registry-replacer/main.go:575]: Right. Currently the assumption is that we will only use this going forward. We should probably make this configurable but we won't need it for older releases.

**Pull request merged**